core/shared/src/main/scala/spire/math/Polynomial.scala

```diff
@@ -490,14 +485,17 @@ with EuclideanRing[Polynomial[C]] with VectorSpace[Polynomial[C], C] {
   override def quotmod(x: Polynomial[C], y: Polynomial[C]): (Polynomial[C], Polynomial[C]) = x /% y
 
   final def gcd(x: Polynomial[C], y: Polynomial[C]): Polynomial[C] = {
-    val result = euclid(x, y)(Polynomial.eq)
+    val result = Gcd.euclid(x, y)(Polynomial.eq, this)
     if (result.degree > 0) {
       result
     } else {
       // return the gcd of all coefficients when there is no higher degree divisor
-      Polynomial.constant(spire.math.gcd(x.coeffsArray ++ y.coeffsArray))
+      val g = Gcd.fromEuclideanRing[C]
```

Comment:
Could we split out `Gcd[Polynomial[C]]` into its own, separate implicit instead? That way we can ask for a `Gcd[C]` instead.

```
class PolynomialGcd[C](implicit ring: EuclideanRing[Polynomial[C]], coeffGcd: Gcd[C]) extends Gcd[Polynomial[C]] {
  ....
}
```